search-java/src/test/java/uk/os/elements/search/android/providers/addresses/AddressesProviderTest.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2016 Ordnance Survey
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.os.elements.search.android.providers.addresses;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import rx.Observable;
+import uk.os.elements.search.android.providers.addresses.service.AddressApi;
+import uk.os.elements.search.android.providers.addresses.service.model.Header;
+import uk.os.elements.search.android.providers.addresses.service.model.Result;
+import uk.os.elements.search.android.providers.addresses.service.model.ServerResponse;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
+public class AddressesProviderTest {
+
+    private final static String API_KEY = "niceTry";
+    private final String BBOX_PARAM = "50.920709,-1.4029539999999998,50.922709,-1.400954";
+    private final String POINT = "50.921709,-1.401954";
+    private final String QUERY = "London";
+    private final String POSTCODE = "SO16 0AS";;
+    private final double USER_LAT = 50.921709;
+    private final double USER_LON = -1.401954;
+
+    private static final Answer<Observable<ServerResponse>> SINGLE_ANSWER = getSingleAnswer();
+
+    @Test
+    public void queryBoundingBox() throws IOException, InterruptedException {
```

Comment:
yup, junit 3 required a 'test' prefix _but_ junit4 doesn't [\[1\]](http://developer.android.com/training/testing/unit-testing/local-unit-tests.html).

Certainly a matter of style.  I note [\[2\]](http://stackoverflow.com/questions/3146821/naming-convention-junit-suffix-or-prefix-test) [\[3\]](http://stackoverflow.com/questions/4491952/prefix-for-testing-methods-in-unit-test-vs-should), where perhaps the test prefix is outdated and could be replaced by behaviour driven _shouldXyz_.

Quick fix is to possibly to prefix with test _but_ longer term perhaps we should create more meaningful test names.  testQueryBoundingBox certainly sounds quite meaningless without seeing the code imo.